### ISSUE TITLE
WGLMakie produces corrupted traces

### ISSUE THREAD

**Issue by pepijndevos** [MakieOrg/Makie.jl#2958]
Here is a MWE
```
using WGLMakie
fig = Figure()

ax = Axis(fig[1, 1])

sg = SliderGrid(
    fig[1, 2],
    ((label = k, range = (10 .^ (-3:0.1:3)), format = "{:.1e}", startvalue = 1)
        for k in ["foo", "bar", "baz"])...,
    width = 350,
    tellheight = false)

sliderobservables = [s.value for s in sg.sliders]
traces = lift(sliderobservables...) do slvalues...
    sleep(rand())
    [sinpi.((0:0.001:100).*val) for val in slvalues]
end

for vec in 1:3
    trace = lift(traces) do sol
        sol[vec]
    end
    lines!(ax, trace)
end

ylims!(ax, -1, 1)

fig
```

And here is the result, note how it sometimes draws filled arcs and other random nonsense

https://github.com/MakieOrg/Makie.jl/assets/168609/0ef91de9-f657-41f5-a4c4-70ca83913fca

**Comment by pepijndevos** [comment]
Something related that isn't captured in this MWE is that the slider track and knob sometimes become disconnected and randomly jump around. Probably also some race condition stuff going on there.

**Comment by SimonDanisch** [comment]
fixed in #3113
